domain/src/main/java/io/homeassistant/companion/android/domain/widgets/WidgetUseCaseImpl.kt

```diff
@@ -0,0 +1,63 @@
+package io.homeassistant.companion.android.domain.widgets
+
+import javax.inject.Inject
+
+class WidgetUseCaseImpl @Inject constructor(
```

Comment:
This is interesting--it's how I originally wrote it but I ended up moving the logic into the "use case" because the term made me think that since this is how I'm USING the repository, I should be putting it here.  I think part of the problem is I don't understand why I am essentially rewriting the same general header three times and the logic once instead of a general method once (plus the interface header) and the specific logic once (plus the interface header).

I don't have any problem making the change, I just want to clarify what is confusing to me about why the code is formatted in this manner.

Also, I just wanted to say thank you for being so patient with me, this is the easily the most complex project I've proposed code for, so this type of code structure is entirely new to me.

Reply:
Changed completed as requested in commit 8b2b9db